sys/arvo.hoon

```diff
@@ -136,8 +138,43 @@
     =+  gen=(rain pax txt)
     =+  rev=(slym (slap bud gen) bud)
     =+  syg=(slym rev arg)
-    ~|  %load-lost
-    +>.^$(vase.vane (slam (slap syg [%limb %load]) (slap rig [%limb %stay])))
+    ::  update the vane itself
+    ::
+    ::    We don't cache the +slap/+slam types because they're only used once
+    ::    right here; they'll never be used again.
+    ::
+    =.  vase.vane
+      ~|  %load-lost
+      (slam (slap syg [%limb %load]) (slap rig [%limb %stay]))
+    ::  prime the new compiler cache
+    ::
+    prime
+  ::  reset and prime the worm cache for scrys
+  ::
+  ::    If the +slap/+slym in scry isn't cached, we spend the majority of
+  ::    the time in a scry in the compiler. The +scry gate cannot have side
+  ::    effects so we can't modify the cache at access time. So we seed the
+  ::    cache with all the things +scry will need when we install the vane
+  ::
+  ++  prime
+    ^+  ..prime
+    ::
+    %_    ..prime
+        worm.vane
+      ::  reset cache and add in vane activation entry
+      ::
+      =^  rig  worm.vane
+        (~(slym wa *worm) vase.vane *[@da @ slyd])
+      ::  cache the access of the %scry arm
+      ::
+      =^  fun  worm.vane  (~(slap wa worm.vane) rig [%limb %scry])
+      ::  cache the call to +mint that the +slym in +scry will do
+      ::
+      =^  pro  worm.vane  (~(mint wa worm.vane) p.fun [%limb %$])
+      ::  cache the call to +mint that the +slot does for the return call
```

Comment:
We don't need to prime this cache if we replace the +slot call in +scry with untyped axis lookup (see other review comment).